### ISSUE TITLE
Markdown links to template class instantiations do not work

### ISSUE THREAD

**Issue by georg-emg** [doxygen/doxygen#8714]
When attempting to create a markdown link to the documentation of a template class specialization, Doxygen creates broken html output. The following link will not appear correctly, for example:

~~~cpp
template <>
struct Test<std::string>;

/// [text](\ref Test<std::string>)
~~~

Links to members of the specialization work correctly. Non-markdown links also work correctly.

**Expected behavior**
Markdown links to template class specializations should produce the same output as conventional links using \ref.

**Screenshots**
![markdown-links-to-template-instantiations](https://user-images.githubusercontent.com/71000124/129180595-41e70485-20bc-47ab-b523-20ed945943a4.png)

**To Reproduce**
This is the project that created the above screenshot:

[markdown-links-to-template-instantiations.zip](https://github.com/doxygen/doxygen/files/6974642/markdown-links-to-template-instantiations.zip)

**Version**
Doxygen version 1.9.1. on  64 bit Windows 10

**Comment by albert-github** [comment]
Note that you are using some none existing settings, (not the cause of your problem):
```
warning: ignoring unsupported tag 'ENABLE_MARKDOWN' at line 1, file Doxyfile
warning: ignoring unsupported tag 'SOURCES' at line 4, file Doxyfile
```

you probably wanted to use `MARKDOWN_SUPPORT` and `INPUT`.

**Comment by doxygen** [comment]
@georg-emg Please verify if the referenced commit fixes the problem.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
